catboost/libs/fstr/shap_values.cpp

```diff
@@ -1235,6 +1441,601 @@ TVector<TVector<double>> CalcShapValues(
     return shapValues;
 }
 
+static inline double GetInteractionEffect(
+    const TVector<TVector<TVector<double>>>& contribsOn,
+    const TVector<TVector<TVector<double>>>& contribsOff,
+    const size_t documentIdx,
+    const int dimension,
+    const int featureIdx
+) {
+    return (contribsOn[documentIdx][dimension][featureIdx] - contribsOff[documentIdx][dimension][featureIdx]) / 2.0;
```

Comment:
for faster performance, change [documentIdx][dimension][featureIdx] to [featureIdx][dimension][documentIdx] when you compute and use contribsOn and contribsOff